src/server/components/lists/helpers.ts

```diff
@@ -11,12 +11,31 @@ import {
   fcdoLawyersPagesByCountry,
   listOfCountriesWithLegalAid,
 } from "server/services/metadata";
+import { URLSearchParams } from "url";
 
 export async function initLists(server: Express): Promise<void> {
   server.use(listsRouter);
 }
 
-export function queryStringFromParams(params: { [name: string]: any }): string {
+/**
+ * To support the select all option for checkbox fields, this function detects if the value "All" is provided and will
+ * remove all other options in the event "Select All" and any additional checkboxes were selected by the user.  Currently
+ * this is only required for lawyers.practiceArea (Areas of law field).
+ * @param params
+ */
+export function preProcessParams(params: { [name: string]: any}): { [name: string]: any} {
+  const paramsCopy = JSON.parse(JSON.stringify(params));
+
+  if (paramsCopy.practiceArea !== undefined) {
+    const areasOfLaw = paramsCopy.practiceArea;
+    if (Array.isArray(areasOfLaw) && areasOfLaw.some(item => item === "All")) {
+      paramsCopy.practiceArea = ["All"];
+    }
+  }
+  return paramsCopy;
+}
```

Comment:
After inspecting params more closely it looks like the value is just a plain comma separated string rather than an array so we can use paramsCopy?.practiceArea?.includes("All") instead.  I've refactored accordingly.